### PR TITLE
Feature/tao 8589/unique field validator

### PR DESCRIPTION
**Pull request by yuryfilippovich**
new ExtendedValidatorInterface for validators which need values another form fields to validate

### CODE REVIEW

File: common/oatbox/validator/ExtendedValidatorInterface.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ */
+namespace oat\oatbox\validator;
+
+interface ExtendedValidatorInterface extends ValidatorInterface
+{
+    /**
+     * @param array $values
+     * @param string $currentPropertyUri
+     */
+    public function populateAdditionValues(array $values, $currentPropertyUri);
```
Comment (augustas) [inline common/oatbox/validator/ExtendedValidatorInterface.php]: I would change the interface to
`public function populateWithFormValues(array $values, $currentPropertyUri);`
But in general I think the solution should be reviewed by an architect, to decide if this is the proper way to pass the modified resource id to a validator.

**Review by augustas — COMMENTED** [review]
Looks okay, review by architect suggested

File: common/oatbox/validator/ExtendedValidatorInterface.php (hunk 1 above)
Comment (zagovorichev) [reply common/oatbox/validator/ExtendedValidatorInterface.php]: totally agree, we don't need this interface

**Review by zagovorichev — CHANGES_REQUESTED** [review]

File: common/oatbox/validator/ExtendedValidatorInterface.php (hunk 1 above)
Comment (yuryfilippovich) [reply common/oatbox/validator/ExtendedValidatorInterface.php]: But if we need to validate some more complex form. For example with name1 and name2 and a least one field should be filled. In this case validation should be aware of other form values

**Pull request closed without merging**